### DISCUSSION TITLE
Wildcard subdomains setup

### DISCUSSION THREAD

**Discussion by skittlesaur** [vercel/community#5078] [Help]
### Summary

I have a domain that is hosted outside of Vercel and i want to use a wildcard subdomain. I have configured the DNS to accept the `TXT` project verification code fine but now I'm stuck in this step.

<img width="892" alt="image" src="https://github.com/vercel/community/assets/61744498/dd58adbf-8431-4e40-8fa6-60481f9562d9">

Enable Vercel DNS button shows a toast with "domain does not exist" _(im guessing because its hosted outside of Vercel domains so it can't really configure it automatically)_. 

How do i setup Vercel's nameservers when it's not giving me any nameservers to use? And does my domain need to be hosted on Vercel Domains for it to work? Also will this break any other subdomains associated with the same domain (baraa.app)?

### Steps to Reproduce

By adding a wildcard subdomain that is has not been purchased from Vercel to some project on Vercel.

**Comment by amyegan** [comment]
~If you share the domain, I can look into this for you~

**Reply by amyegan** [reply]
I just noticed that the domain is in the screenshot. Thanks for sharing that!

I see the domain associated with a different GitHub account. Since you said you already added `TXT` to verify the domain, you must have already set up `A` and `CNAME` records or changed nameservers. Could you please let me know the name or a deployment link of the project this domain was added to?

**Reply by skittlesaur** [reply]
Hi, the domain im trying to add is `*.builder.baraa.app` and the one in the screenshot is associated with this GitHub account since its the only one i have and it's configured in a project for this account.

I had nameservers set up but i reverted to the original Porkbun nameservers since it caused issues in my other projects until the domain shows up in Vercel's domains tab and i will reconfigure them since some of them relate to my bachelor and i need them online.

The project i'm trying to set wildcard subdomain for is called `webbuilder-preview` and Vercel's deployment link is [webbuilder-preview.vercel.app](https://webbuilder-preview.vercel.app/).

**Reply by amyegan** [reply]
Thanks for sharing that extra info. I see this subdomain associated with your project with valid configuration. 

Since you're using non-Vercel nameservers, you can use this `NS` record to add a wildcard subdomain: [vercel.com/guides/wildcard-domain-without-vercel-nameservers](https://vercel.com/guides/wildcard-domain-without-vercel-nameservers)

**Reply by skittlesaur** [reply]
Thank you, I have checked this when looking for a solution but I don't see my domain on Vercel's domains list though so I can't really continue with the second step.

<img width="1272" alt="image" src="https://github.com/vercel/community/assets/61744498/a69241fe-3528-40f5-b7c2-ffd9fb0b2729">

This is also an issue when I tried using Vercel's nameserver since I can't configure my DNS on vercel once i move it there.

**Reply by amyegan** [reply]
You should be able to click the "add" button on that page and follow the steps in that workflow to re-add the `builder.baraa.app` subdomain to your project. Do you get an error when you try that?

**Reply by skittlesaur** [reply]
Yes, `Cannot add *.builder.baraa.app since it's already in use by your account.`

Also deleting the domain from the project and setting it up from this domains page doesn't change much. It just redirects me to the project domain settings page with the domain added, but it doesn't show in the main dashboard domains page.

**Reply by amyegan** [reply]
Can you try just `builder.baraa.app` to get that subdomain associated with your project and account? Then you should be able to add the wildcard alongside the regular subdomain in project settings.

**Reply by skittlesaur** [reply]
I did and it added the domain in the project settings. Still the domain is not showing in `/dashboard/domains` so I can't really enable Vercel DNS.

<img width="941" alt="image" src="https://github.com/vercel/community/assets/61744498/658fc542-e79d-4570-a9f8-659c6a481d9e">
<img width="1492" alt="image" src="https://github.com/vercel/community/assets/61744498/0db6bd27-3303-4417-8eba-0b7172e38883">

**Reply by amyegan** [reply]
Thank you for bearing with me as we work to figure this out. I just tried a few things and got my own wildcard subdomain working. Here’s what I had to do using my testing domain, `amyegan.dev`:

1. Added `NS` records for `_acme_challenge.wild` based on the [wildcard subdomains guide](https://vercel.com/guides/wildcard-domain-without-vercel-nameservers).
2. Added `NS` records for `wild.amyegan.dev` to use Vercel nameservers.
3. Removed all `_vercel` verification `TXT` records to avoid conflicts.
4. Added `wild.amyegan.dev` and `*.wild.amyegan.dev` to a project and enable Vercel DNS. 

Configuration showed as invalid for a moment while the records caught up with my rapid changes. I waited a few minutes and clicked the “Refresh” button, then everything showed “Valid Configuration”. The subdomains work as expect and I see the domain in my Domains list. 🎉

If you're not able to change the root domain nameservers, please give these steps a try and let me know if you still have trouble with it.

**Reply by skittlesaur** [reply]
I have done these and waiting till DNS propagates but still I can't see the domain in `/dashboard/domains` nor I can enable Vercel DNS. Clicking the Enable DNS button shows this error still.

<img width="1440" alt="image" src="https://github.com/vercel/community/assets/61744498/91f77061-31c9-45e4-b8fc-92e730e00602">

<img width="995" alt="image" src="https://github.com/vercel/community/assets/61744498/21cf5a14-f817-42e9-8301-e83cb20fa6e7">

I noticed that my provider attaches `.baraa.app` at the end of the records value; could that be the issue?
